### DISCUSSION TITLE
Support for embedding vectors

### DISCUSSION THREAD

**Discussion by carlosbaraza** [meilisearch/product#642] [Feedback & Feature Proposal]
It would be great if neural embeddings could be added to each item and weight the embedding neighbours as part of the search.

This could potentially enable:
- More meaningful searches with support for synonyms, etc.
- Getting a few results to prompt a model like GPT-4 with some context (langchain style)

Even more ambitious and cool would be to somehow allow integrating a ML conversational API like OpenAI ChatGPT API. Not sure if it would be a good add-on for meilisearch, but it could somehow merge both crazy speeds of Meilisearch with the relevancy of a conversational AI.

Any ideas?

**Comment by oluademola** [comment]
Hello @carlosbaraza 👋

Thank you for your feedback 🙂 We have given this feature some thought and have started a discussion [here](https://github.com/meilisearch/product/discussions/621) about it; we'd appreciate your help and experience in designing it. To ensure that your feedback is directed to the appropriate channel, I encourage you to let us know your thoughts and ideas in detail as a comment in the [discussion](https://github.com/meilisearch/product/discussions/621). This will help us to address your feedback more efficiently.

Thanks in advance 🙏

**Comment by macraig** [comment]
Closing in favor of https://github.com/meilisearch/product/discussions/621
